mars-rover/mars_rover.rb

```diff
@@ -1,3 +1,121 @@
+require_relative 'rover_coordinate'
+
+$NORTH = 'n'
+$EAST  = 'e'
+$SOUTH = 's'
+$WEST  = 'w'
+$STEP_FORWARD  = 'f'
+$STEP_BACKWARD = 'b'
+$TURN_RIGHT    = 'r'
+$TURN_LEFT     = 'l'
+
 class MarsRover
+  attr_accessor :obstacle
+
+  def initialize (current_coordinate,direction,grid)
+    @current_coordinate = current_coordinate
+    @facing = RoverDirection.new(direction)
+    @grid = grid
+  end
+
+  def get_position
+    [@current_coordinate.to_s,@facing.to_s]
+  end
+
+  def execute commands
+    @obstacle = nil
+
+    commands.downcase.each_char do |command|
+      cmd=RoverCommand.new(command)
+      if cmd.step_command? then
+        delta_coordinate = RoverCoordinate.coordinate_from_vector(@facing.to_s)
```

Comment:
The to_s is an indication that there's something wrong. Perhaps the direction class needs to include the relevant constants?